src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -119,7 +119,9 @@ internal partial class Binder
             ConstantValue constantValue = this.FoldConstantConversion(syntax, source, conversion, destination, diagnostics);
             if (conversion.Kind == ConversionKind.DefaultOrNullLiteral && source.Kind == BoundKind.DefaultExpression)
             {
-                source = ((BoundDefaultExpression)source).Update(constantValue, destination);
+                var defaultExpression = (BoundDefaultExpression)source;
+                Debug.Assert(defaultExpression.TargetType == null);
```

Comment:
>defaultExpression.TargetType [](start = 29, length = 28)

Why bother asserting this instead of simply using ```defaultExpression.TargetType``` in the update below. That would be correct no matter what its value is.

Reply:
The default conversion is simply not defined for a default expression that provides an explicit target type.  This is asserting that fact.

---
In reply to: [274638373](https://github.com/dotnet/roslyn/pull/34889#discussion_r274638373) [](ancestors = 274638373)